src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -529,7 +529,14 @@ async fn healthcheck(
                     Err(HealthcheckError::GroupNameError.into())
                 }
             }
-            None => Err(HealthcheckError::NoLogGroup.into()),
+            None => {
+                if config.create_missing_group.unwrap() == true {
+                    info!("Cloudwatch group_name will be created at runtime");
```

Comment:
Seeing this written out, I'm actually coming around to the idea that we should just skip the healthcheck completely if `create_missing_group` is `true` similar to how we skip it if `log_group` is a template. In either case, the health check just always returns true.

Reply:
Like add an `or create_missing_group == true` at this point?

Reply:
Well, I might have two separate conditionals to output two different log messages, but otherwise yeah.

Reply:
The only reason I didn't want to completely short-circuit there was since the config defaults to true, it would make the healthcheck somewhat "pointless" but looking at the possible ways the healthcheck fails they're pretty much all related to the group name being problematic

Reply:
🤔 yeah, that's true. I suppose this would still check that the AWS credentials can access the `DescribeLogGroups` API though we skip even that currently if the `group_name` is dynamic. I think we should have it be consistent for both. That is:

* If either `log_group` name is dynamic or `create_missing_group` is true, then check the API permissions, but otherwise don't fail
* If either `log_group` name is dynamic or `create_missing_group` is true, then skip the check entirely

My preference would be for the former.

Reply:
Agreed - I'll take some time this afternoon to try larger rewrite